skops/io/_general.py

```diff
@@ -405,6 +407,23 @@ def object_get_state(obj: Any, save_context: SaveContext) -> dict[str, Any]:
     except Exception:
         pass
 
+    # Then we check if the output of __reduce__ is of the form
+    # (constructor, (constructor_args,))
+    # If the constructor is the same as the object's type, then we consider it
+    # safe to call it with the specified arguments.
+
+    reduce_output = obj.__reduce__()
+    # note that we do "=="" to compare types instead of "is", since we only accept
+    # exact matches here.
+    if len(reduce_output) == 2 and reduce_output[0] == type(obj):
+        return {
+            "__class__": obj.__class__.__name__,
```

Comment:
For clarity, should we use `reduce_output[0].__name__` or `type(obj).__name__` here (i.e. one of the two objects used above)?

Reply:
using `type(obj).__name__` now